druid/src/widget/widget_ext.rs

```diff
@@ -303,16 +310,20 @@ mod tests {
         let widget = Slider::new()
             .background(Color::BLACK)
             .align_left()
-            .border(Color::BLACK, 1.0);
+            .border(Color::BLACK, 1.0)
+            .foreground(Color::WHITE);
         assert!(widget.border_is_some());
         assert!(!widget.background_is_some());
+        assert!(!widget.foreground_is_some());
```

Comment:
For this test case, you should remove the foreground call, because you're testing for there to be no foreground.

Reply:
What you're saying makes sense, but what is going on with the background check here? Not really a problem with this PR, but looks weird.

Reply:
The one group of tests makes sure that fields that weren't set stay unset, and the other ones ensures that ones that were set stay set.

Reply:
Aha, it's a double use of `Container`. The background is applied to the inner `Container` but the assert is for the outer `Container`. So the `foreground` method call should not be removed, but instead be moved earlier in the callhain, before `align_left`.

Reply:
Just moved that. I think it is ready for another round of comments, if any! I will do another pass but I think all the tests should pass now